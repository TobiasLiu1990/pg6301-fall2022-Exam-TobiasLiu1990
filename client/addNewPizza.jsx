import { fetchJSON } from "./fetchJson";
import React, { useState } from "react";

function FormInput({ label, value, onChangeValue }) {
  return (
    <>
      <label>
        <strong>{label}</strong>{" "}
        <input value={value} onChange={(e) => onChangeValue(e.target.value)} />
      </label>
      <br></br>
    </>
  );
}

export function AddNewPizza({ pizzaApi }) {
  const [pizza, setPizza] = useState("");
  const [price, setPrice] = useState("");
  const [ingredient, setIngredient] = useState("");
  const [allergen, setAllergen] = useState("");

  let ingredients = [];
  let allergens = [];

  ingredients = ingredient.split(" ");
  allergens = allergen.split(" ");

  async function handleSubmit(e) {
    e.preventDefault();

    pizzaApi.addPizza({ pizza, price, ingredients, allergens })

    await fetchJSON("/api/menu/new", {
      method: "post",
      json: { pizza, price, ingredients, allergens },
    });
  }

  return (
    <div>
      <h1>Add your new pizza</h1>

      <form onSubmit={handleSubmit}>
        <FormInput label={"Pizza: "} value={pizza} onChangeValue={setPizza} />
        <FormInput label={"Price: "} value={price} onChangeValue={setPrice} />
        <FormInput
          label={"Ingredients (Separate by space): "}
          value={ingredient}
          onChangeValue={setIngredient}
        />
        <FormInput
          label={"Allergen (Separate with space): "}
          value={allergen}
          onChangeValue={setAllergen}
        />

        <button>Submit Pizza</button>
      </form>
    </div>
  );
}
