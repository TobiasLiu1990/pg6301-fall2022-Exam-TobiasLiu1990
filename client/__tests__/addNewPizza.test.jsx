import ReactDOM from "react-dom";
import React from "react";
import { act, Simulate } from "react-dom/test-utils";
import { AddNewPizza } from "../addNewPizza";

describe("add pizza component", () => {
  it("should show pizza form", async () => {
    const element = document.createElement("div");
    await act(async () => ReactDOM.render(<AddNewPizza />, element));

    expect(element.innerHTML).toMatchSnapshot();

    const inputLabels = Array.from(
      element.querySelectorAll("form label strong")
    ).map((label) => label.innerHTML);

    expect(inputLabels).toEqual(["Pizza: ", "Price: ", "Ingredients (Separate by space): ", "Allergen (Separate with space): "])
  });

  it('should add a pizza', async () => {
    const addPizza = jest.fn();

    const element = document.createElement("div");
    await act(async () => ReactDOM.render(<AddNewPizza pizzaApi={{ addPizza }}/>))

  });
});