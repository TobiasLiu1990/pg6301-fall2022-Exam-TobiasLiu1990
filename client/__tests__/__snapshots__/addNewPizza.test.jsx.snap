// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add pizza component should show pizza form 1`] = `"<div><h1>Add your new pizza</h1><form><label><strong>Pizza: </strong> <input value=""></label><br><label><strong>Price: </strong> <input value=""></label><br><label><strong>Ingredients (Separate by space): </strong> <input value=""></label><br><label><strong>Allergen (Separate with space): </strong> <input value=""></label><br><button>Submit Pizza</button></form></div>"`;
